app/Http/Controllers/Admin/RoleController.php

```diff
@@ -0,0 +1,358 @@
+            })
+
+            ->addColumn('action', function ($permission) use ($role) {
+                $flag = DB::table('permission_role')
+                        ->where('role_id', $role->id)
+                        ->where('permission_id', $permission->id)
+                        ->where('deleted_at', null)
+                        ->first();
+
+                if ($flag != null) {
+                    $string = '<input type="hidden" id="checked-'. $permission->id .'" value="1">';
+
+                    $string = $string . '<i id="action-'. $permission->id. '"
+                                            class="fa fa-check-circle"
+                                            data-tooltip="tooltip"
+                                            title="Xóa"
+                                            onclick="updatePermission('. $role->id . ', '. $permission->id .')"
+                                            aria-hidden="true"
+                                            style="cursor: pointer; color: #3598dc; font-size: 20px;">
+                                        </i>';
+                } else {
+                    $string = '<input type="hidden" id="checked-' .$permission->id . '" value="0">';
+
+                    $string = $string . '<i id="action-'. $permission->id .'"
+                                            class="fa fa-circle-notch"
+                                            data-tooltip="tooltip"
+                                            title="Thêm"
+                                            onclick="updatePermission('. $role->id .', '. $permission->id .')"
+                                            aria-hidden="true"
+                                            style="cursor: pointer; color: #3598dc; font-size: 20px;">
+                                        </i>';
+                }
+
+                return $string;
+            })
+
+        ->make(true);
+    }
+
+    /**
+     * [updatePermissionRole : Cập nhật quyền hạn cho vai trò]
+     * @param  Request $request [description]
+     * @return [type]           [description]
+     */
+    public function updatePermissionRole(Request $request)
+    {
+        if ($request->checked == 1) {
+            DB::table('permission_role')
+                ->where('role_id', $request->role_id)
+                ->where('permission_id', $request->permission_id)
+                ->delete();
+
+            return response()->json([
+                'error' => false,
+                'message' => 'deleted',
+            ], 200);
+        } else {
+            DB::table('permission_role')->insert([
+                'role_id' => $request->role_id,
+                'permission_id' => $request->permission_id,
+                'created_at' => date('Y-m-d H:m:s', time()),
+            ]);
+
+            return response()->json([
+                'error' => false,
+                'message' => 'added',
+            ], 200);
+        } 
+    }
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/RoleController.php</br>Whitespace found at end of line